charms/kserve-controller/src/charm.py

```diff
@@ -0,0 +1,314 @@
+                field_manager=self._lightkube_field_manager,
+                template_files=K8S_RESOURCE_FILES,
+                context=self._context,
+                logger=log,
+            )
+        return self._k8s_resource_handler
+
+    @property
+    def _controller_pebble_layer(self):
+        """Return the Pebble layer for the workload."""
+        return Layer(
+            {
+                "services": {
+                    self._controller_container_name: {
+                        "override": "replace",
+                        "summary": "KServe Controller",
+                        "command": "/manager --metrics-addr=:8080",
+                        "startup": "enabled",
+                        "environment": {
+                            "POD_NAMESPACE": self.model.name,
+                            "SECRET_NAME": "kserve-webhook-server-cert",
+                        },
+                    },
+                }
+            }
+        )
+
+    @property
+    def _rbac_proxy_pebble_layer(self):
+        """Return the Pebble layer for the workload."""
+        return Layer(
+            {
+                "services": {
+                    self._rbac_proxy_container_name: {
+                        "override": "replace",
+                        "summary": "Kube Rbac Proxy",
+                        "command": "/usr/local/bin/kube-rbac-proxy --secure-listen-address=0.0.0.0:8443 --upstream=http://127.0.0.1:8080 --logtostderr=true --v=10",
+                        "startup": "enabled",
+                    }
+                }
+            }
+        )
+
+    def _on_kserve_controller_ready(self, event):
+        """Define and start a workload using the Pebble API.
+
+        Learn more about Pebble layers at https://github.com/canonical/pebble
+        """
+        try:
+            self.gen_certs(self.model.name, self.app.name)
+
+            self.controller_container.push("/tmp/k8s-webhook-server/serving-certs/tls.crt", Path("/run/cert.pem").read_text(), make_dirs=True)
+            self.controller_container.push("/tmp/k8s-webhook-server/serving-certs/tls.key", Path("/run/server.key").read_text(), make_dirs=True)
+
+            update_layer(
+                self._controller_container_name,
+                self.controller_container,
+                self._controller_pebble_layer,
+                log,
+            )
+        except ErrorWithStatus as e:
+            self.model.unit.status = e.status
+            if isinstance(e.status, BlockedStatus):
+                log.error(str(e.msg))
+            else:
+                log.info(str(e.msg))
+
+        # TODO determine status checking if rbac proxy is also up
+        self.unit.status = ActiveStatus()
+
+    def _on_kube_rbac_proxy_ready(self, event):
+        """Define and start a workload using the Pebble API.
+
+        Learn more about Pebble layers at https://github.com/canonical/pebble
+        """
+        try:
+            update_layer(
```

Comment:
Same here with container connection check.

Reply:
chisme will [check](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/pebble/_update_layer.py#L22) that